corehq/apps/userreports/views.py

```diff
@@ -699,6 +699,18 @@ def export_data_source(request, domain, config_id):
     for sql_filter in params.sql_filters:
         q = q.filter(sql_filter)
 
+    # xls format has limit of 65536 rows
+    # First row is taken up by headers
+    if params.format == 'xls' and q.count() >= 65535:
+        keyword_params = dict(**request.GET)
+        keyword_params.update(format='xlsx')
```

Comment:
couchexport has constants for these that are used in most of the code over the strings (even though that's a little silly)